front/lib/models/plan.ts

```diff
@@ -125,7 +134,7 @@ export class Subscription extends Model<
   declare updatedAt: CreationOptional<Date>;
 
   declare sId: string; // unique
-  declare status: "active" | "ended";
+  declare status: "active" | "ended" | "processing" | "cancelled";
```

Comment:
What's the difference between ended and cancelled ?
Do we really need processing?

Reply:
Ended will be when we ended a subscription, and cancelled when the user has started a checkout processed and cancelled it without subscribing. I prefer making the difference between both use cases!

Reply:
I thiiink we won't create subscription before the user is fully subscribed and we received a webhook no?

Reply:
But maybe we will want to do that in which case I understand the "processing"

Reply:
But we won't need that. We'll get a webhook but also the possibility to pull the subscription once the user is redirected back, so I think we don't need these two statuses no?

Reply:
Sorry I was writing as I was thinking:

- I think we don't want to create a subscription object before the user is fully subscribed
- We have two places to do so:
  - as the user is redirected back
  - as we receive a webhook from stripe (if the user dropped the redirect)
  - both will be idempotent

I think for this reason we don't want "pending" subscriptions or "cancelled" subscription as this simplified a lot our modeling relying on Stripe webhooks as source of truth and on the redirect back as a shortcut to make sure we're up to date as the user comes back.

Does that make senes?

Reply:
Yes if it has no value knowing about users who dropped in the middle of the checkout, we can def do that! Adding a commit!

Reply:
Yes no value in DB (this will be part of our analytics whatever that means 👍)

Reply:
@spolu ready for another look whenever you can 🙇🏻‍♀️